common/hmac.c

```diff
@@ -0,0 +1,36 @@
+#include <bitcoin/privkey.h>
+#include <ccan/array_size/array_size.h>
+#include <ccan/mem/mem.h>
+#include <common/hmac.h>
+#include <wire/wire.h>
+
+void hmac(const void *src, size_t slen,
+	  const void *key, size_t klen,
```

Comment:
Seeing the previous cleanup would we ever want to use anything other than a `struct secret`?

Reply:
Sigh, unfortunately the hmac construction used by Sphinx uses the string as the *key* not src. :(

Found this out when I tried cleaning up the first time.